openapi-bal-service/src/main/java/io/ballerina/openapi/converter/service/OpenAPIComponentMapper.java

```diff
@@ -719,54 +719,107 @@ private ArraySchema handleArray(int arrayDimensions, Schema property, ArraySchem
     }
 
     /**
-     * This util uses to set the constraint value for relevant schema field.
+     * This util uses to set the integer constraint values for relevant schema field.
      */
-    private Schema setConstraintValueToSchema(ConstraintAnnotation constraintAnnot, Schema property) {
-
-        if (property instanceof ArraySchema) {
-            property.setMaxItems(constraintAnnot.getMaxLength().isPresent() ?
-                    Integer.valueOf(constraintAnnot.getMaxLength().get()) : null);
-            property.setMinItems(constraintAnnot.getMinLength().isPresent() ?
-                    Integer.valueOf(constraintAnnot.getMinLength().get()) : null);
-        } else {
-            property.setMaxLength(constraintAnnot.getMaxLength().isPresent() ?
-                    Integer.valueOf(constraintAnnot.getMaxLength().get()) : null);
-            property.setMinLength(constraintAnnot.getMinLength().isPresent() ?
-                    Integer.valueOf(constraintAnnot.getMinLength().get()) : null);
-        }
+    private Schema setIntegerConstraintValuesToSchema(ConstraintAnnotation constraintAnnot, Schema properties) {
+        BigDecimal minimum = null;
+        BigDecimal maximum = null;
+        if (constraintAnnot.getMinValue().isPresent()) {
+            minimum = BigDecimal.valueOf(Integer.parseInt(constraintAnnot.getMinValue().get()));
+        } else if (constraintAnnot.getMinValueExclusive().isPresent()) {
+            minimum = BigDecimal.valueOf(Integer.parseInt(constraintAnnot.getMinValueExclusive().get()));
+            properties.setExclusiveMinimum(true);
+        }
+
+        if (constraintAnnot.getMaxValue().isPresent()) {
+            maximum = BigDecimal.valueOf(Integer.parseInt(constraintAnnot.getMaxValue().get()));
+        } else if (constraintAnnot.getMaxValueExclusive().isPresent()) {
+            maximum = BigDecimal.valueOf(Integer.parseInt(constraintAnnot.getMaxValueExclusive().get()));
+            properties.setExclusiveMaximum(true);
+        }
+        properties.setMinimum(minimum);
+        properties.setMaximum(maximum);
+        return properties;
+    }
 
+    /**
+     * This util uses to set the number (float, double) constraint values for relevant schema field.
+     */
+    private Schema setNumberConstraintValuesToSchema(ConstraintAnnotation constraintAnnot, Schema properties) {
```

Comment:
Rather than catching `ParseException` within this method lets just throw the exception from here and capture it in the `setConstraintValueToSchema`. Then the exception handling would be consistent.